lib/tasks/unpublish.rake

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+namespace :unpublish do
```

Comment:
Another non blocker - do you think we could avoid having the task as unpublish? just wonder if we can try avoid that terminology and we've mostly managed it here so far.

Since everything applies to document what do you think of a namespace of document then we can have `document:retire` which seems to mean the same as `unpublish:retire_document` but is shorter and avoids that dreaded word 😉

Reply:
Well, they do all call the DocumentUnpublishingService, so we aren't really avoiding that words. I couldn't think of another word that groups the tasks.

Reply:
You could have DocumentRemover and DocumentRetirer and not group them?

Reply:
sorry! not joking but your call on it - definitely don't want to suggest something that'll take a load of time and not add much value